src/guest_memory.rs

```diff
@@ -513,10 +513,20 @@ pub trait GuestMemory {
         self.find_region(addr).map(|_| addr)
     }
 
-    /// Returns the address plus the offset if it is present within the memory of the guest.
+    /// Returns the address plus the offset if the whole range is present within the memory of
```

Comment:
This is actually a breaking change in terms of `checked_offset` semantics. If the new functionality is helpful, we should just create a new method (`checked_range` or `checked_range_offset` or smt along those lines).